### PR TITLE
style: developer guide

### PR DESCRIPTION
**Pull request by henryiii**
- chore: activate more developer guidelines
- style: pre-commit run -a for pyupgrade
- style: fix flake8 issues
- fix: correct all flake8 warnings

### CODE REVIEW

File: decaylanguage/dec/dec.py [hunk 1]
```diff
@@ -435,10 +432,10 @@ def _add_decays_to_be_copied(self):
                 copied_decay = match.__deepcopy__(None)
                 copied_decay.children[0].children[0].value = decay2copy
                 copied_decays.append(copied_decay)
-            except:
+            except Exception:
```
Comment (henryiii) [inline decaylanguage/dec/dec.py:435]: No bare excepts! These might be able to be tightened up further, but at least now they don't catch memory errors, control C, etc.

File: decaylanguage/dec/dec.py [hunk 2]
```diff
@@ -714,14 +711,14 @@ def print_decay_modes(
...
-        l.sort(key=operator.itemgetter(0), reverse=(not ascending))
+        ls.sort(key=operator.itemgetter(0), reverse=(not ascending))
 
-        for bf, info in l:
-            print("%12g : %s" % (bf, info))
+        for bf, info in ls:
+            print("{:12g} : {}".format(bf, info))
 
-    def build_decay_chains(self, mother, stable_particles=[]):
+    def build_decay_chains(self, mother, stable_particles=()):
```
Comment (henryiii) [inline decaylanguage/dec/dec.py:721]: Never use a mutable structure in an argument default, it's highly bug-prone. (Note: many of these comments are probably to myself, as I wrote quite a few of theses...)

File: decaylanguage/dec/decparser.py [hunk 3]
```diff
@@ -200,25 +195,25 @@ def addline(self, line):
...
-        except:
-            raise Exception("Cannot parse decay line: %s" % line)
-        ##        mod_found = False
-        ##         for x in known_models:
-        ##             if x in line:
-        ##                 mod_found = True
-        ##         if not mod_found:
-        ##             print '\n', line
+        except Exception:
+            raise RuntimeError("Cannot parse decay line: %s" % line)
```
Comment (henryiii) [inline decaylanguage/dec/decparser.py:199]: Always should be a bit more specific than "Exception" when throwing. RuntimeError is a pretty good "I don't know I want to throw" default.

File: tests/decay/test_decay.py [hunk 4]
```diff
@@ -146,7 +146,7 @@ def test_DecayMode_charge_conjugate():
     assert "Decay model: TAUHADNU [-0.108, 0.775, 0.149, 1.364, 0.4]" in dm.describe()
 
     dd = DaughtersDict("pi- pi0 nu(tau)")
-    dd.charge_conjugate(pdg_name=True) == DaughtersDict("pi+ pi0 nu(tau)~")
+    assert dd.charge_conjugate(pdg_name=True) == DaughtersDict("pi+ pi0 nu(tau)~")
```
Comment (henryiii) [inline tests/decay/test_decay.py:149]: This test tested nothing. I love this bugbear check :)

File: tests/dec/test_dec.py [hunk 5]
```diff
@@ -187,14 +185,14 @@ def test_global_photos_flag():
     p = DecFileParser(DIR / "../data/defs-aliases-chargeconj.dec")
     p.parse()
 
-    assert p.global_photos_flag() == True
+    assert p.global_photos_flag()
```
Comment (henryiii) [inline tests/dec/test_dec.py:188]: Never use `==` with None, True, or False.

File: decaylanguage/dec/decparser.py (hunk 3 above)
Comment (eduardo-rodrigues) [reply decaylanguage/dec/decparser.py:199]: Yep.

**Comment by henryiii** [comment]
After this one, I could move to a src/ structure, then we would pretty much be fully following the dev guidelines, I think. :)

File: tests/dec/test_dec.py (hunk 5 above)
Comment (eduardo-rodrigues) [reply tests/dec/test_dec.py:188]: Ah, did not know that. OK.

File: decaylanguage/dec/dec.py [hunk 6]
```diff
@@ -189,9 +186,9 @@ def parse(self, include_ccdecays=True):
         # Retrieve all info on the default Lark grammar and its default options,
         # effectively loading it
         opts = self.grammar_info()
-        extraopts = {
-            k: v for k, v in opts.items() if k not in ("lark_file", "parser", "lexer")
-        }
+        # extraopts = {
+        #     k: v for k, v in opts.items() if k not in ("lark_file", "parser", "lexer")
+        # }  not used later?
```
[the remark this answers is not in the document]
Comment (eduardo-rodrigues) [reply decaylanguage/dec/dec.py:191]: Right now I'm really lacking time so let me check this PR as a couple of things may need discussion ...

File: decaylanguage/dec/dec.py (hunk 2 above)
Comment (eduardo-rodrigues) [reply decaylanguage/dec/dec.py:721]: Oh, all these functions for building decay chains are my babies, actually, don't worry. I do tend to use lists as lot even when tuples are just fine, and this one is an example. Indeed the user is meant to give a "list" of particles they want to declare as stable for the purpose of playing with all the info parsed.

Comment (henryiii) [reply decaylanguage/dec/dec.py:721]: As a good rule, lists, sets, and dicts should never be default arguments. Too easy to modify across calls. So swapping for tuple or frozenset is an easy fix. Dicts are harder, have to use None and check for it.

File: decaylanguage/dec/dec.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (eduardo-rodrigues) [reply decaylanguage/dec/dec.py:191]: I guess you used flake8 to find this. Great. In fact it was meant to be used just below in the code when instantiating `Lark`:
`parser = Lark(self.grammar(), parser=opts["parser"], lexer=opts["lexer"])`
should be
`parser = Lark(self.grammar(), parser=opts["parser"], lexer=opts["lexer"], **extraopts)`.
Can you update within your PR?

File: decaylanguage/dec/dec.py (hunk 1 above)
Comment (eduardo-rodrigues) [reply decaylanguage/dec/dec.py:435]: Good catch. This was indeed overlooked.

**Review by eduardo-rodrigues — APPROVED** [review]
I left a tiny update to make. Otherwise great!

File: decaylanguage/dec/dec.py [hunk 7]
```diff
@@ -191,10 +188,10 @@ def parse(self, include_ccdecays=True):
         opts = self.grammar_info()
         extraopts = {
             k: v for k, v in opts.items() if k not in ("lark_file", "parser", "lexer")
-        }
+        }  not used later?
```
Comment (eduardo-rodrigues) [inline decaylanguage/dec/dec.py:191]:
```suggestion
        }
```

**Pull request merged**